src/Entities/Player.cpp

```diff
@@ -89,6 +89,12 @@ cPlayer::cPlayer(cClientHandlePtr a_Client, const AString & a_PlayerName) :
 	m_UUID((a_Client != nullptr) ? a_Client->GetUUID() : ""),
 	m_CustomName("")
 {
+	if (a_PlayerName.empty() || a_PlayerName.length() > 16)
+	{
+		LOGERROR("Playername of %s is too long(>16) or empty", a_PlayerName.c_str());
+		VERIFY(false);
+	}
```

Comment:
I don't think there has any problems

> 在 2016年1月13日，16:20，bibo38 <notifications@github.com> 写道：
> 
> In src/Entities/Player.cpp:
> 
> > @@ -89,6 +89,12 @@ cPlayer::cPlayer(cClientHandlePtr a_Client, const AString & a_PlayerName) :
> >  	m_UUID((a_Client != nullptr) ? a_Client->GetUUID() : ""),
> >  	m_CustomName("")
> >  {
> > +	if (a_PlayerName.empty() || a_PlayerName.length() > 16)
> > +	{
> > +		LOGERROR("Playername of %s is too long(>16) or empty", a_PlayerName.c_str());
> > +		VERIFY(false);
> > +	}
> Asserting something on an input variable is a not a good practise, so I only asserted it in debug mode with VERIFY and gave an fatal error in the log. In my opinion this should never happen, but I don't want to just crash the server because somebody created an invalid cPlayer for something.
> 
> —
> Reply to this email directly or view it on GitHub.
>